### PR TITLE
Pareto distribution

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
split from #1177

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

File: cupy/random/_kernels.py [hunk 1]
```diff
@@ -252,6 +252,20 @@
 }
 '''
 
+rk_standard_exponential_definition = '''
+__device__ double rk_standard_exponential(rk_state *state) {
+    /* We use -log(1-U) since U is [0, 1) */
+    return -log(1.0 - rk_double(state));
+}
+'''
```
Comment (toslunar) [inline cupy/random/_kernels.py]: The kernel is already in the file.

File: cupy/random/generator.py [hunk 2]
```diff
@@ -198,6 +198,21 @@ def normal(self, loc=0.0, scale=1.0, size=None, dtype=float):
...
+        .. seealso::
+            :func:`cupy.random.pareto_kernel` for full documentation,
+            :meth:`numpy.random.RandomState.pareto`
+        """
+        a = cupy.asarray(a)
+        if size is None:
+            size = a.shape
+        y = cupy.empty(shape=size, dtype=dtype)
+        _kernels.pareto_kernel(a, self.rk_seed, y)
+        self.rk_seed += numpy.prod(size)
```
Comment (toslunar) [inline cupy/random/generator.py]: Similarly to #1420, this distribution could be generated from the result of `self._random_sample_raw`.

File: cupy/random/distributions.py [hunk 3]
```diff
@@ -230,6 +230,32 @@ def normal(loc=0.0, scale=1.0, size=None, dtype=float):
     return x
 
 
+def pareto(a, size=None, dtype=float):
+    """Pareto II or Lomax distribution.
+
+    Returns an array of samples drawn from the pareto II distribution. Its
```
Comment (toslunar) [inline cupy/random/distributions.py]: `:s/pareto/Pareto/`?

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

File: cupy/random/distributions.py (hunk 3 above)
Comment (toslunar) [reply cupy/random/distributions.py]: Could you fix this?

File: cupy/random/distributions.py [hunk 4]
```diff
@@ -283,6 +283,32 @@ def normal(loc=0.0, scale=1.0, size=None, dtype=float):
...
+    probability density function is defined as
+
+    .. math::
+        f(x) = \\alpha(1+x)^{-(\\alpha+1)},
+
+    Args:
+        a (float): Parameter of the gamma distribution :math:`\\alpha`.
```
Comment (toslunar) [inline cupy/random/distributions.py]: Could you fix a typo?
`gamma` → `Pareto`

File: cupy/random/distributions.py [hunk 5]
```diff
@@ -283,6 +283,32 @@ def normal(loc=0.0, scale=1.0, size=None, dtype=float):
...
+        size (int or tuple of ints): The shape of the array. If ``None``, a
+            zero-dimensional array is generated.
```
Comment (toslunar) [inline cupy/random/distributions.py]: It seems `a` can be a `cupy.ndarray` and `size=None` will generate an output with the same size as `a`.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**